packages/transform/src/lib/TransformMiddleware.ts

```diff
@@ -0,0 +1,485 @@
+import {
+  ApplyMiddlewareOptions,
+  isEveryByHookPayload,
+  isEveryByValuePayload,
+  isMapByHookPayload,
+  isMapByPathPayload,
+  isSomeByHookPayload,
+  isSomeByValuePayload,
+  JoshMiddleware,
+  Method,
+  Payload,
+  PostProvider,
+  PreProvider,
+  resolveVersion,
+  Semver
+} from '@joshdb/provider';
+import { Awaitable, objectToTuples } from '@sapphire/utilities';
+
+@ApplyMiddlewareOptions({ name: 'transform' })
+export class TransformMiddleware<BeforeValue = unknown, AfterValue = unknown> extends JoshMiddleware<
+  // @ts-expect-error 2322 - 'ContextData<BeforeValue, AfterValue>' has properties in common with type 'Context', ts thinks they are incompatible
+  TransformMiddleware.ContextData<BeforeValue, AfterValue>,
+  AfterValue
+> {
+  public get version(): Semver {
+    return resolveVersion('[VI]{version}[/VI]');
+  }
+
+  @PreProvider()
+  public async [Method.Each]<ReturnValue = BeforeValue>(payload: Payload.Each<ReturnValue>): Promise<Payload.Each<ReturnValue>> {
+    const { after } = this.context;
+    const hook = async (value: AfterValue, key: string) => {
+      return payload.hook((await after(value, key, null)) as ReturnValue, key) as ReturnValue;
+    };
+
+    await this.provider[Method.Each]({ ...payload, hook });
+
```

Comment:
Metadata to prevent /core from also running Each is required here

Reply:
Hmmm, for `Method.Each` should I return the non-transformed values and emitWarning them?